### PR TITLE
Feature/108 performance issue related to datapoint name

### LINKED ISSUE #108: Performance issue related to datapoint name
## TODO/DONE

- [ ] Test with number of calls
- [ ] Move datapointName update in questionField

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE
Fix performance issue by :

- [x] Removing useEffect for timer
- [x] Removing useEffect for setDatapointName
- [x] Removing useEffect in TypeGeo when getting current location
- [x] Removing useEffect in TypeCascade when set initialValues
- [x] Refactoring dataPointName

### Additional updates

- [x] Fix subtitle position and text-overflow
- [x] Cascade name appears twice in the data point name 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205171411395760
  - https://app.asana.com/0/0/1205178220298796

### CODE REVIEW

File: app/src/pages/__tests__/FormPageContinueSubmissionAndSubmit.test.js [hunk 1]
```diff
@@ -323,7 +325,7 @@ describe('FormPage continue saved submision then submit', () => {
           6: ['Traveling'],
           7: ['Fried Rice'],
         },
-        duration: 0.15, // in minutes
+        duration: 27235390, // in seconds
```
Comment (dedenbangkit) [inline app/src/pages/__tests__/FormPageContinueSubmissionAndSubmit.test.js]: Remember that server should received xxx minutes for the duration

File: app/src/pages/__tests__/FormPageHandleSaved.test.js [hunk 2]
```diff
@@ -329,11 +343,16 @@ describe('FormPage handleOnSaveForm', () => {
...
+      FormState.update((s) => {
+        s.surveyStart = 1634123456789;
+      });
+      fireEvent.press(saveButtonElement);
+    });
 
     await waitFor(() => {
       expect(crudDataPoints.saveDataPoint).toHaveBeenCalledWith({
-        duration: 0,
+        duration: 27235390,
```
Comment (dedenbangkit) [inline app/src/pages/__tests__/FormPageHandleSaved.test.js]: should be in minutes, ceil up

File: app/src/pages/__tests__/FormPageHandleSubmitted.test.js [hunk 3]
```diff
@@ -283,7 +285,7 @@ describe('FormPage handleOnSubmitForm', () => {
     // save datapoint to database
     await waitFor(() => {
       expect(crudDataPoints.saveDataPoint).toHaveBeenCalledWith({
-        duration: 0.15, // in minutes
```
Comment (dedenbangkit) [inline app/src/pages/__tests__/FormPageHandleSubmitted.test.js:286]: revert this

File: app/src/pages/__tests__/FormPageContinueSubmissionAndSave.test.js [hunk 4]
```diff
@@ -329,7 +332,7 @@ describe('FormPage continue saved submision then save', () => {
     await waitFor(() => {
       expect(crudDataPoints.saveDataPoint).not.toHaveBeenCalled();
       expect(crudDataPoints.updateDataPoint).toHaveBeenCalledWith({
-        duration: 0,
+        duration: 27235390,
```
Comment (dedenbangkit) [inline app/src/pages/__tests__/FormPageContinueSubmissionAndSave.test.js]: when we convert this to minutes?

**Review by dedenbangkit — CHANGES_REQUESTED** [review]
Hi @ifirmawan note that server API received minutes, otherwise it won't work. For other functions / states, lets discuss tomorrow.

File: app/src/pages/__tests__/FormPageContinueSubmissionAndSave.test.js (hunk 4 above)
Comment (ifirmawan) [reply app/src/pages/__tests__/FormPageContinueSubmissionAndSave.test.js]: actually before call saveDatapoints
https://github.com/akvo/nmis-mobile/blob/feature/108-performance-issue-related-to-datapoint-name/app/src/pages/FormPage.js#L181-L185

let me try to fix the mock test

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**